src/ResourceParser.h

```diff
@@ -475,10 +494,11 @@ namespace snowcrash {
         }
 
         /** Check Parameters eligibility in URI template */
+        template<typename T>
         static void checkParametersEligibility(const MarkdownNodeIterator& node,
                                                const SectionParserData& pd,
                                                Parameters& parameters,
-                                               const ParseResultRef<Resource>& out) {
+                                               const ParseResultRef<T>& out) {
```

Comment:
I have thought about that. I think this function won't be getting specialised in the future. Firstly, this is a helper function which is restricted to a small logic block and the logic in the function won't change for any data structures thus ruling out specialised functions.